src/types/index.ts

```diff
@@ -592,6 +592,22 @@ export class StarknetContract {
         return this.adaptOutput(functionName, executed.stdout.toString());
     }
 
+    /**
+     * Computes message fee estimation
+     * @param {string} fromAddress - Address of message source
+     * @param {string} functionName - Function name for entry point selector
+     * @param {Array<Numeric>} payload - Payload to send
+     * @returns Fee estimation
+     */
+    async estimateMessageFee(fromAddress: string, functionName: string, payload: Numeric[]) {
+        return this.hre.starknetWrapper.estimateMessageFee(
```

Comment:
Also, we could check if the function is indeed an `@l1_handler`

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/312/commits/63eb11f0d8a5c569cb8e2526f18f209b71bc5471